Duplicati/Server/webroot/ngax/scripts/directives/backupEditUri.js

```diff
@@ -245,6 +253,82 @@ backupApp.directive('backupEditUri', function(gettextCatalog) {
+                $scope.oauth_service_link = oauthUrl;
+            }
+            const token = ur.searchParams.get("authid");
+            if (token != null) $scope.oauth_create_token = token;
+            scope.oauth_start_link = scope.oauth_service_link + '?type=' + scope.Backend.Key + '&token=' + scope.oauth_create_token;
+
+            $scope.oauth_in_progress = true;
+
+            var w = 450;
+            var h = 600;
+
+            var url = $scope.oauth_start_link;
+
+            var countDown = 100;
+            var ft = $scope.oauth_create_token;
+            var left = (screen.width / 2) - (w / 2);
+            var top = (screen.height / 2) - (h / 2);
+            var wnd = window.open(url, '_blank', 'height=' + h + ',width=' + w + ',menubar=0,status=0,titlebar=0,toolbar=0,left=' + left + ',top=' + top)
+
+            // check the OAuth server
+            var recheck = function () {
+                countDown--;
+                if ((countDown > 0) && ft == $scope.oauth_create_token) {
+                    $http.jsonp($scope.oauth_service_link + 'fetch?callback=JSON_CALLBACK', {params: {'token': ft}}).then(
+                        function (response) {
+                            if (response.data.authid) {
+                                $scope.AuthID = response.data.authid;
+                                $scope.oauth_in_progress = false;
+                                if (wnd != null) {
+                                   wnd.close();
+                                   wnd = null;
+                                }
+                                countDown = 0;
+                            } else {
+                                if (wnd != null && wnd.closed)
+                                    wnd = null;
+                                else
+                                    setTimeout(recheck, 3000);
+                            }
+                        },
+                        function (response) {
+                            if (response.status == 404) {
+                                $scope.oauth_in_progress = false;
+                                if (wnd != null) {
+                                   wnd.close();
+                                   wnd = null;
+                                }
+                               countDown = 0;
+                            } else
+                                setTimeout(recheck, 3000);
+                        }
+                    );
+                } else {
+                    $scope.oauth_in_progress = false;
+                    if (wnd != null) {
+                        wnd.close();
+                        wnd = null;
+                    }
+                }
+            };
+            if (wnd != null)
+                setTimeout(recheck, 6000);
+            return false;
+        };
+
+        $scope.oauth_start_token_creation = function () {
+            builduri(perform_oauth_start_token_creation);
```

Comment:
`builduri` will fail in case some validation is not successful. That means you need to specify the authid for some backends before you can create an auth ID. It would be better to use `scope.AdvancedOptions` to check for the `oauth-url` option instead. That has a binding to the options editor, so it should always be up to date without needing to build the uri.